zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -162,6 +162,38 @@ public void deleteAuthorizationsByOwnerTypeAndIdPrefix(
         });
   }
 
+  @Override
+  public void update(final long authorizationKey, final AuthorizationRecord authorization) {
+    this.authorizationKey.wrapLong(authorizationKey);
+    final var persistedAuthorization =
+        authorizationByAuthorizationKeyColumnFamily.get(this.authorizationKey);
+
+    if (persistedAuthorization == null) {
+      // if there are no persisted authorization, we should not update the authorization
+      return;
+    }
+
+    // remove the old permissions
+    persistedAuthorization
+        .getPermissions()
+        .forEach(
+            permissionType -> {
+              removePermission(
+                  persistedAuthorization.getOwnerType(),
+                  persistedAuthorization.getOwnerId(),
+                  persistedAuthorization.getResourceType(),
+                  permissionType,
+                  Set.of(persistedAuthorization.getResourceId()));
+            });
+
+    // delete the old authorization record
+    authorizationByAuthorizationKeyColumnFamily.deleteExisting(this.authorizationKey);
+    permissionKeyByAuthorizationKeyColumnFamily.deleteExisting(this.authorizationKey);
+
+    // create the new authorization record
+    create(authorizationKey, authorization);
```

Comment:
❓ Why delete and recreate instead of doing an update?

Reply:
Because I think this is the easiest approach, the persisted authorization can be easily updated. However to handle correctly the permissions we need to first remove them from the related record and re-add the new ones.

Reply:
I see, so mainly to ensure the permissions are added. I can live with that.